Documentation/Addition of Required Documentation (#7)
gopavasanth/WhatsApp-Karo: CONTRIBUTING.md

# Contribution Rules📚:

> - You are allowed to make pull requests. We just merge it ;)

Reviewer: We don't just merge it. We review and then merge. So, you can remove that sentence and just keep Anyone is allowed to make a PR.

- Do NOT remove other content.

> - Styling/code can be pretty, ugly or stupid, big or small as long as it works

Reviewer: Sorry, but code has to follow some standards. Formatting and style of writing code has to be good. "as long as it works" is not acceptable. Can you please update.

> - Try to keep pull requests small to minimize merge conflicts

Reviewer: A PR can be big. It just needs to fit in one feature. So, I think rephrasing is needed.

> 

Reviewer: I think we need to include that to work on projects, Flutter is needed. (Not exactly at this place but somewhere).

## Getting Started 🤩🤗:

- Fork this repo (button on top)

- Clone on your local machine

```terminal
git clone https://github.com/<yourusername>/WhatsApp-Karo.git
```
- Navigate to project directory.

```terminal
cd WhatsApp-Karo
```
## For PR Request 
- Create a new Branch

```markdown
git checkout -b "< your_new_branch_name >"
```
*your branch name should follow some naming convention* : `issue_name/changes`

- Add your changes to your branch

> ```markdown

Reviewer: For all commands related, you can keep sh instead of markdown

# Contributing to WhatsApp-Karo💕

Thank you for your interest in contributing to WhatsApp-Karo! We welcome contributions from the community to help improve and grow this project.

## Contribution Rules📚:

- You are allowed to make pull requests. Just make sure it adds some value to the project.

> - Do NOT remove other content.

Reviewer: ```suggestion
- Do NOT remove other content unless necessary.
```

> - Styling/code should follow some standards.

Reviewer: ```suggestion
- Styling/code should follow community standards (eg. proper code formatting).
```

> - PRs can be Big or Small. It will be reviewed and then only it will be merged/accepted.

Reviewer: ```suggestion
- PR's can be Big or Small. Don't be afraid to open a PR, you will always get help.
```

```
- Navigate to project directory.

```sh
cd WhatsApp-Karo
```
## For PR Request 
- Create a new Branch

```sh
git checkout -b "< your_new_branch_name >"
```
*your branch name should follow some naming convention* : `issue_name/changes`

- Add your changes to your branch

```sh
git add .
```
- Commit your changes with some `Relevent Messages`.

```sh
git commit -m "Relevant message"
```
- Then push 
```sh
git push origin my-new-branch
```

- Create a new pull request from your forked repository

<br>

## Avoid Conflicts {Syncing your fork}

- An easy way to avoid conflicts is to add an 'upstream' for your git repo, as other PR's may be merged while you're working on your branch/fork.   

```sh
git remote add upstream 
```

- You can verify that the new remote has been added by typing
```sh
git remote -v
```

- To pull any new changes from your parent repo simply run
```sh
git merge upstream/master
```

This will give you any eventual conflicts and allow you to easily solve them in your repo. It's a good idea to use it frequently in between your own commits to make sure that your repo is up to date with its parent.

## Setup Guide

> This app is written in Flutter. Even though Flutter is cross-platform, This app is currently focused on Android or iOS. So in order to setup and run the project locally, you need to have an emulator(android or iOS) or a real device.

Reviewer: ```suggestion
This app is written in Flutter. Even though Flutter is cross-platform, This app is currently focused on Android or iOS. So in order to setup and run the project locally, you need to have an emulator(android or iOS) or a real device. And more importantly, flutter installed.
```

It's just a suggestion, feel free to keep it how it looks good.
